src/libstd/sys_common/memchr.rs

```diff
@@ -65,15 +66,12 @@ pub mod fallback {
         let usize_bytes = mem::size_of::<usize>();
 
         // search up to an aligned boundary
-        let align = (ptr as usize) & (usize_bytes- 1);
-        let mut offset;
-        if align > 0 {
-            offset = cmp::min(usize_bytes - align, len);
+        let mut offset = unsafe { align_offset(ptr as *const _, usize_bytes) };
```

Comment:
Should this be using the inherent methods added in this PR?

Reply:
The compiler complained that it doesn't know the unstable feature (stage0). This would become a mess of cfgs, so I opted to not do this here.

Reply:
Sounds like a stage0 inherent method on pointers which does the old method via casts and not(stage0) uses the intrinsic?